external/fv3fit/fv3fit/reservoir/utils.py

```diff
@@ -12,3 +16,54 @@ def _square_evens(v: np.ndarray) -> np.ndarray:
 
 def square_even_terms(v: np.ndarray, axis=1) -> np.ndarray:
     return np.apply_along_axis(func1d=_square_evens, axis=axis, arr=v)
+
+
+def get_ordered_X(X_mapping, variables):
```

Comment:
If this is more than a helper function, I would use variable name `X' with type hints here instead of `X_mapping`.